***day 1 week 2 in training***

***1-importing pandas and numpy and clear the data from titanic***

using numpy because its faster

In [2]:
import pandas as pd
import numpy as np
titanic=pd.read_csv(r"C:\Users\Hp\Desktop\binx_mohammadabuhamed\binx_mohammadabuhamed\week2\day1\tested.csv")
age=titanic['Age']
age= age.dropna()
npage=np.array(age)

***2- Use NumPy and Pandas to calculate:***

- **Mean:** The average value of the dataset.
- **Median:** The middle value when the data is sorted.
- **Mode:** The value that appears most frequently in the dataset.
- **Standard Deviation:** The square root of the variance. It measures how far the data values are spread from the mean.
- **Interquartile Range (IQR):** The difference between Q3 and Q1. It measures the spread of the middle 50% of the data.

In [3]:
print(np.mean(npage))
print(np.median(npage))
print(age.mode())
print(np.std(age))
q1,q3=np.percentile(age,[25,75])
IQR=q3-q1
print(IQR)

30.272590361445783
27.0
0    21.0
1    24.0
Name: Age, dtype: float64
14.159835886272559
18.0


***3- Which is better here, the mean or the median?***

In this dataset, I chose the mean because it is very close to the median. This indicates that there are no significant outliers affecting the data, so the mean is a good representation of a typical value.

***4-Summary***

The average age of passengers is 30.27 years, while the median age is 27 years.

The most common ages are 21 and 24 years.

The standard deviation is 14.16 years, indicating that passenger ages have a moderate spread around the mean.

The IQR is 18 years, which means the middle 50% of passenger ages are spread across a range of 18 years.
